Nevis system Frame Rollover fix

ibsafa opened the pull request:
### Description

Introduces two fixes: 
1) Correction to the GPS timestamp issued by NTB. When nevis clock frames roll over in between a PPS pulse and a trigger, the assigned trigger timestamp is ambiguous for one second. this fix ensures that 1 second of data is timestamped correctly. 
2) NTB and TPC boardreaders introduce a correction to the frame number (adding multiples of 2^24 when to account for rollovers)

### Testing details
- *Run number associated with test*: 
- Other information:

Commit:
timestamp fix for frame rollover

Commit:
iterate frame if rollover occurs

Commit:
apply correction more than once

ibsafa commented:
We have determined that we don't need PR#113 in sbndaq-artdaq-core, and just board reader changes are enough to introduce the fixes we want. I will commit to this PR one more time and then it will be ready for review. will update with a comment here when that's done.

Commit:
updates to remove artdaq_core changes

ibsafa commented:
Made changes to remove artdaq-core changes. @afilkins @eyandel This PR is now ready for review. I believe test runs are planned for today. Thank you!

gennadiy-fnal commented:
@ibsafa  There are merge conflicts that need to be addressed.

ibsafa committed:
Merge branch 'release/v1_10_03' into bug/isafa_NTB_Frame_Rollover

ibsafa commented:
@gennadiy-fnal fixed it, thanks!

Commit:
added trace to track frame number -- for testing

Commit:
Merge branch 'bug/isafa_NTB_Frame_Rollover' of github.com:SBNSoftware/sbndaq-artdaq into bug/isafa_NTB_Frame_Rollover

afilkins commented on sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc:
```diff
@@ -252,16 +257,53 @@ bool sbndaq::NevisTB_generatorBase::FillNTBFragment(artdaq::FragmentPtrs &frags,
+    }
+
+    corrFrame = tframe + rollCounter*16777216;
+    TLOG(TLVL_INFO) << " Corrected Frame:  " << corrFrame <<  " Uncorrected Frame: " << tframe;
```
Can we change this to Debug? it spams the log messages pretty heavily

ibsafa replied:
would I still be able to see it in the log files?

afilkins replied:
It should, but another option would be to make it a TRACE instead which would show up in logs but not the message viewer

ibsafa replied:
done

afilkins commented on sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc:
```diff
@@ -252,16 +257,53 @@ bool sbndaq::NevisTB_generatorBase::FillNTBFragment(artdaq::FragmentPtrs &frags,
+    int32_t   tolerance  = 16770000;
+    int32_t   corrFrame;
+
+    TLOG(TLVL_INFO) << "Frame difference (between GPS and trigger): " << frame_diff;
```
DEBUG for this one too?

ibsafa replied:
done. for this one I just commented out. it's not needed anymore

ibsafa committed:
Update NevisTB_generatorBase.cc

ibsafa committed:
Update NevisTPC_generatorBase.cc

ibsafa reviewed (commented):
@afilkins I changed the traces to DEBUG as requested. thanks for looking at this!

ibsafa committed:
Update NevisTB_generatorBase.cc

ibsafa committed:
Update NevisTPC_generatorBase.cc

afilkins committed:
update printout

use the actual set variable value in the printout in case there is a no data timeout
